Core/Core/Inbox/MessageDetails/Model/MessageDetailsInteractor.swift

```diff
@@ -0,0 +1,32 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2023-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Combine
+
+public protocol MessageDetailsInteractor {
+    // MARK: - Outputs
+    var state: CurrentValueSubject<StoreState, Never> { get }
```

Comment:
Since these subjects can never fail we could use `CurrentValueRelay` from the `CombineExt` framework. The declaration will be cleaner without the `Never` part. You can send a value to it with `accept()` instead of `send()`.

Reply:
we use it this way in other interactors, so if we want to change it, we should do it globally, not just in this single case.